scipy/signal/tests/test_signaltools.py

```diff
@@ -1920,6 +1919,30 @@ def test_numpy_fastpath(self):
         assert_allclose(correlate(a, b, mode='valid'), [32])
 
 
+@pytest.mark.parametrize("mode", ["valid", "same", "full"])
+@pytest.mark.parametrize("behind", [True, False])
+def test_correlation_lags(mode, behind):
+    # generate random data
+    rng = np.random.RandomState(0)
+    in1 = rng.standard_normal(1000)
+    # generate offset version of array to correlate with
+    if behind:
+        # y is behind x
+        in2 = np.concatenate([rng.standard_normal(100), in1])
+        expected = -100
+    else:
+        # y is ahead of x
+        in2 = in1[100:]
+        expected = 100
+    # cross correlate, returning lag information
+    correlation = correlate(in1, in2, mode=mode)
+    lags = correlation_lags(in1.size, in2.size, mode=mode)
+    # identify the peak
+    lag_index = np.argmax(correlation)
```

Comment:
Can you test for more than just the argmax. Perhaps check `lags.shape == correlation.shape` and I would also use more than 1 size of inputs.

Reply:
I've added a few different lengths as parameterised inputs and added some variables to reflect that. Also added the test you suggest.

Reply:
You need to test an odd length since you have `if in1_len % 2 == 0` in the code.

Reply:
Sure thing.